git-publish

```diff
@@ -246,9 +247,12 @@ def git_send_email(to_list, cc_list, patches, suppress_cc, in_reply_to, dry_run=
     args += ['--confirm=never']
     args += patches
     if dry_run:
-            return _git_with_stderr(*args[1:])[0]
+        return _git_with_stderr(*args[1:])[0]
     else:
-        if subprocess.call(args) != 0:
```

Comment:
It looks like this patch replaces subprocess.call() in order to UTF-8 decode output from git.

Why does subprocess.call() not work?

Does it work if you run the same command-line from your command prompt?

Reply:
It's result invalid encoding output in powershell, so I capture the output first, then print it out.

Reply:
Is the output encoding is invalid when you run the same git command-line directly from PowerShell (without git-publish)?

Reply:
I noticed that git uses console APIs when it detects a console (instead of a pipe or text file as stdout/stderr):
https://github.com/git/git/blob/master/compat/winansi.c#L610

Maybe it's possible to invoke subprocess.call() in a way that allows git to detect PowerShell and use the console API (they convert from git's internal UTF-8 encoding to Windows wide characters).
https://docs.python.org/3/library/subprocess.html

But decoding git's output is fine too.